### PR TITLE
feat(run): support -- passthrough for app arguments

### PR DESCRIPTION
**Pull request by mattleibow**
## Summary

Adds support for the `--` passthrough separator in `winapp run`, so users can pass flags to their launched app without escaping:

```sh
# Before (requires escaping)
winapp run . --args "--my-flag value"

# After (no escaping needed)
winapp run . -- --my-flag value
```

Tokens after `--` are collected and merged with any `--args` value. Unknown tokens *before* `--` are still caught and reported as errors (e.g. typos like `--detch`).

`tool` and `store` already used `TreatUnmatchedTokensAsErrors = false` — no changes needed for those.

## Changes

- **`RunCommand.cs`**: Set `TreatUnmatchedTokensAsErrors = false`, collect post-`--` tokens, count-based pre-dash validation, merge with `--args`
- **`RunCommandTests.cs`**: 14 tests covering parse-level behaviour, handler happy paths, rejection, and mode interactions (`--no-launch`, `--detach`, `--debug-output`)
- **`docs/usage.md`** + **`docs/fragments/skills/winapp-cli/setup.md`**: Updated examples

## Test coverage

| Area | Tests |
|---|---|
| Parse: bare `--`, no parse errors for unknown pre-dash tokens | 3 |
| Handler: bare `--`, multiple args, merge with `--args`, space-quoting | 4 |
| Handler rejection: unknown option, same-value duplicate bug case | 3 |
| Mode interaction: `--no-launch`, `--detach`, `--debug-output` | 3 |

All 57 RunCommand tests pass.

### CODE REVIEW

**Comment by mattleibow** [comment]
## Note: `ToolCommand` argument escaping

`ToolCommand.cs` currently uses a naive space-quoting approach for tool arguments:
```csharp
Arguments = string.Join(" ", toolArgs.Select(a => a.Contains(' ') ? $"\"{a}\"" : a))
```

This has the same embedded-quotes / trailing-backslash issue documented in earlier review comments. The correct fix is to switch to `ProcessStartInfo.ArgumentList`:
```csharp
foreach (var arg in toolArgs)
    processStartInfo.ArgumentList.Add(arg);
```

`ProcessStartInfo.ArgumentList` (available since .NET 5) lets the BCL handle `CommandLineToArgvW`-compatible escaping internally — there's no need to call `WindowsCommandLine.EscapeArgument` since we're launching a real process (not going through COM).

This was intentionally left out of the current PR to keep scope focused on the `run` command passthrough feature. Tracked as a follow-up improvement.

File: src/winapp-CLI/WinApp.Cli/Commands/RunCommand.cs [hunk 1]
```diff
@@ -132,6 +138,65 @@ public override async Task<int> InvokeAsync(ParseResult parseResult, Cancellatio
...
 
+            // Collect tokens after '--' as pass-through arguments to the launched application.
+            // This lets users write: winapp run . -- --flag value
+            // instead of having to escape: winapp run . --args "--flag value"
+            var passthroughArgs = new List<string>();
+            var seenDoubleDash = false;
+            foreach (var token in parseResult.Tokens)
+            {
+                if (token.Type == TokenType.DoubleDash) { seenDoubleDash = true; continue; }
+                if (seenDoubleDash) { passthroughArgs.Add(token.Value); }
```
[the remark this answers is not in the document]
Comment (mattleibow) [reply src/winapp-CLI/WinApp.Cli/Commands/RunCommand.cs]: Fixed — this was one of the first issues caught after the initial review. The implementation was subsequently refactored: token collection logic now lives in `WindowsCommandLine.ExtractPassthroughArgs()` which iterates tokens and only treats the *first* `DoubleDash` token as a separator. Any additional `--` tokens encountered after `seenDoubleDash = true` are forwarded as literal arguments. This is covered by the test `ExtractPassthroughArgs_LiteralDoubleDashAfterSeparator_IsForwarded` in `WindowsCommandLineTests.cs`.

File: src/winapp-CLI/WinApp.Cli/Commands/RunCommand.cs [hunk 2]
```diff
@@ -132,6 +138,65 @@ public override async Task<int> InvokeAsync(ParseResult parseResult, Cancellatio
...
+                        logger.LogError("{UISymbol} Unrecognized option or argument: '{Token}'. Check the option name and try again.", UiSymbols.Error, bad);
+                    }
+                    return 1;
+                }
+            }
+
+            // Merge '--args' value with any tokens collected after '--'.
+            if (passthroughArgs.Count > 0)
+            {
+                var passthroughStr = string.Join(" ", passthroughArgs.Select(a => a.Contains(' ') ? $"\"{a}\"" : a));
```
[the remark this answers is not in the document]
Comment (mattleibow) [reply src/winapp-CLI/WinApp.Cli/Commands/RunCommand.cs]: Fixed — rather than patching the inline quoting, we extracted a dedicated `WindowsCommandLine` helper class (`src/winapp-CLI/WinApp.Cli/Helpers/WindowsCommandLine.cs`) with a `EscapeArgument()` method that implements the full `CommandLineToArgvW`-compatible escaping algorithm: empty args become `""`, backslash runs before embedded quotes are doubled and a backslash is added, trailing backslashes before the closing quote are doubled. `RunCommand` now calls `WindowsCommandLine.JoinArguments()` which applies `EscapeArgument()` to every token. The correctness of the algorithm is verified by round-trip tests in `WindowsCommandLineTests.cs` that use P/Invoke `CommandLineToArgvW` to confirm the encoded string round-trips back to the original values.

File: src/winapp-CLI/WinApp.Cli.Tests/RunCommandTests.cs [hunk 3]
```diff
@@ -809,4 +809,213 @@ public async Task RunCommand_DetachWithoutJson_DoesNotOutputJson()
...
+        var command = GetRequiredService<RunCommand>();
+
+        var exitCode = await ParseAndInvokeWithCaptureAsync(command,
+            [_tempDirectory.FullName, "--", "--title", "hello world"]);
+
+        Assert.AreEqual(0, exitCode, "Command should succeed");
+        Assert.AreEqual("--title \"hello world\"", _fakeAppLauncherService.LaunchCalls[0].Arguments,
+            "Passthrough args containing spaces should be quoted");
+    }
+
```
[the remark this answers is not in the document]
Comment (mattleibow) [reply src/winapp-CLI/WinApp.Cli.Tests/RunCommandTests.cs:927]: Added — both edge cases are now covered:

1. **Literal `--` forwarding** — `ExtractPassthroughArgs_LiteralDoubleDashAfterSeparator_IsForwarded` in `WindowsCommandLineTests.cs` and `RunCommand_DoubleDashPassthrough_LiteralDoubleDash_ForwardedToApp` in `RunCommandTests.cs` verify that `winapp run . -- --` forwards `--` to the app.

2. **Embedded quotes and backslashes** — The `WindowsCommandLineTests.cs` file has a full suite of escaping tests including round-trip P/Invoke verification for embedded quotes, trailing backslashes, and the combination. `RunCommand_DoubleDashPassthrough_ValueWithSpace_QuotedInLaunchArgs` in `RunCommandTests.cs` covers the integration path where a space-containing value reaches the launcher correctly quoted.

File: src/winapp-CLI/WinApp.Cli/Commands/RunCommand.cs [hunk 4]
```diff
@@ -93,6 +94,11 @@ static RunCommand()
 
     public RunCommand() : base("run", "Creates packaged layout, registers the Application, and launches the packaged application.")
     {
+        // Allow unmatched tokens so that 'winapp run . -- --flag value' passes '--flag value'
+        // to the launched app without escaping. Unknown options before '--' are caught manually
+        // in the handler so typo detection still works.
+        TreatUnmatchedTokensAsErrors = false;
+
```
[the remark this answers is not in the document]
Comment (mattleibow) [reply src/winapp-CLI/WinApp.Cli/Commands/RunCommand.cs]: Good catch — this is a real regression. `OptionTypoValidator.FindLikelyLongOptionTypo` explicitly bails out when `TreatUnmatchedTokensAsErrors = false` (line 29), so `winapp run . -manifest foo` was silently downgraded to a generic "Unrecognized option" message.

Fixed in a8ce9d4. Rather than duplicating the option-name collection logic inline, I added `OptionTypoValidator.TryGetTypoSuggestion(string token, CommandResult)` — a single-token form of the same check. The `RunCommand` handler already iterates unknown tokens via `ExtractPassthroughArgs`; it now calls `TryGetTypoSuggestion` for each and emits "Did you mean '--manifest'?" when the pattern matches, falling back to the generic message otherwise.

Note: the `Validators.Add` constructor approach (as suggested) can't see unmatched tokens — `CommandResult.Tokens` only contains matched tokens; unmatched ones live in `ParseResult.UnmatchedTokens` which isn't reachable from a validator callback. The handler already has `parseResult` access so that's the natural home for this check.

Added 3 tests: `ParseOptions_SingleDashTypo_NoParseError_HandlerValidates`, `RunCommand_SingleDashTypo_ReturnsError`, and `ParseOptions_SingleDashTypo_AfterDoubleDash_NoError` (verifies that `-manifest` after `--` is treated as a passthrough arg, not a typo).

[the remark this answers is not in the document]
Comment (mattleibow) [reply src/winapp-CLI/WinApp.Cli/Commands/RunCommand.cs]: Further updated in 840c72a with a cleaner architecture. Rather than using TreatUnmatchedTokensAsErrors = false (which disables OptionTypoValidator) or adding a duplicate TryGetTypoSuggestion, the fix uses Argument<string[]> PassthroughArgument with ArgumentArity.ZeroOrMore. System.CommandLine natively routes all post--- tokens into this argument, so TreatUnmatchedTokensAsErrors remains at its default — the OptionTypoValidator path in Program.cs is fully preserved.

One caveat: ZeroOrMore also absorbs unrecognised option-like tokens before -- as positionals. The handler detects this via WindowsCommandLine.SplitPassthroughTokens(tokens, allAbsorbed), which diffs GetValue(PassthroughArgument) (all absorbed) against a post--- token walk using count-based matching. Anything not accounted for in the post-dash walk is rejected. TryGetTypoSuggestion and ExtractPassthroughArgs have both been removed.

**Review by nmetulev — APPROVED** [review]
